### PR TITLE
Add containter_security_context Closes #493

### LINKED ISSUE #493: Add container_security_context query for Kubernetes
### Platform
Kubernetes

### Description
Add a new "Container Security Context" query, that checks if a container has the security context applied

### PR DESCRIPTION
**Pull request by rafaela-soares**
Since security context should be applied in pods and containers, I thought in two cases:

- Pod has not a security context applied

- All or some containers have not a security context applied

Closes #493

### CODE REVIEW

File: assets/queries/k8s/container_security_context/query.rego [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package Cx
+
+CxPolicy [ result ] {
+    document := input.document[i]
+    specInfo := getSpecInfo(document)
```
Comment (joelsou5a) [inline assets/queries/k8s/container_security_context/query.rego:5]: Add metadata information to get a more specific searchKey
metadata := document.metadata

File: assets/queries/k8s/container_security_context/query.rego [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    
+    spec := document.spec
+    
+    document.kind == "Pod"
+    not spec.securityContext
+
+	result := {
+                "documentId": 		  input.document[i].id,
+                "issueType":		  "MissingAttribute",
+                "searchKey": 	      sprintf("%s", [specInfo.path]),
```
Comment (joelsou5a) [inline assets/queries/k8s/container_security_context/query.rego:15]: "searchKey": sprintf("metadata.name=%s.%s", [metadata.name, specInfo.path]),

File: assets/queries/k8s/container_security_context/query.rego [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    
+    lengthContainers := count(containers)
+    
+    count({x | containers[x]; object.get(containers[x], "securityContext", "undefined") != "undefined" }) != lengthContainers
+
+
+	result := {
+                "documentId": 		  input.document[i].id,
+                "issueType":		  "MissingAttribute",
+                "searchKey": 	      sprintf("%s.containers", [specInfo.path]),
```
Comment (joelsou5a) [inline assets/queries/k8s/container_security_context/query.rego:36]: "searchKey": sprintf("metadata.name=%s.%s.containers", [metadata.name, specInfo.path]),

File: assets/queries/k8s/container_security_context/query.rego [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                "issueType":		  "MissingAttribute",
+                "searchKey": 	      sprintf("%s", [specInfo.path]),
+                "keyExpectedValue":   sprintf("Pod has a security context applied in document[%d]",[i]),
+                "keyActualValue": 	  sprintf("Pod has not a security context applied in document[%d]",[i])
+              }
+}
+
+
+CxPolicy [ result ] {
+    document := input.document[i]
```
Comment (joelsou5a) [inline assets/queries/k8s/container_security_context/query.rego:23]: Add metadata information to get a more specific searchKey
metadata := document.metadata

**Review by joelsou5a — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/container_security_context/query.rego [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    specInfo := getSpecInfo(document)
+
+    metadata := document.metadata
+
+    containers := document.spec.containers
+    
+    lengthContainers := count(containers)
+    
+    count({x | containers[x]; object.get(containers[x], "securityContext", "undefined") != "undefined" }) != lengthContainers
```
Comment (joelsou5a) [inline assets/queries/k8s/container_security_context/query.rego:35]: You could iterate over the containers and specify each of them are failing.

File: assets/queries/k8s/container_security_context/metadata.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "container_security_context",
+  "queryName": "Container Security Context",
```
Comment (joaomartinscx) [inline assets/queries/k8s/container_security_context/metadata.json:3]: Query name should identify the problem with the resource, like in this case, if the container or Pod doesn't have a security context defined

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/without_security_context/query.rego [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    document.kind == "Pod"
+    
+    object.get(spec, "securityContext", "undefined") == "undefined"
+
+	result := {
+                "documentId": 		  input.document[i].id,
+                "issueType":		  "MissingAttribute",
+                "searchKey": 	      sprintf("metadata.name=%s.%s", [metadata.name, specInfo.path]),
+                "keyExpectedValue":   sprintf("metadata.name=%s has a security context",[specInfo.path]),
+                "keyActualValue": 	  sprintf("metadata.name=%s does not have a security context",[specInfo.path])
```
Comment (joaomartinscx) [inline assets/queries/k8s/without_security_context/query.rego:20]: I think you meant **metadata.name** instead of **specInfo.path** here

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Review by joaomartinscx — COMMENTED** [review]
Everything looks good to me, I recommend approval and merge of this PR.

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**